lsp/nls/src/field_walker.rs

```diff
@@ -347,6 +372,21 @@ impl<'a> FieldResolver<'a> {
                     Vec::new()
                 }
             }
+            Term::Array(_, attrs) => attrs
+                .pending_contracts
```

Comment:
I'm surprised that there's anything in `pending_contracts` at this point. If I recall correctly, `pending_contracts` on array attributes is only filled at evaluation type when applying an `Array` contract, which updates this field. It's a runtime thing. Do you get anything out of it here in practice?

Reply:
Hm, I'm not actually sure. I'll check if removing it changes anything.

Reply:
@jneem let's proceed with this version, code structure is less important. However, maybe we should flag this now with either a comment or remove it altogether (at worse re-add it later if it turns out to do something in practice, but I doubt it)